tools/topology/topology2/cavs/platform/intel/dmic-generic.conf

```diff
@@ -84,6 +97,8 @@ Object.PCM {
 			name $DMIC0_PCM_CAPS
 			# only 32-bit capture supported now
 			formats 'S32_LE'
+			channels_min $NUM_DMICS
```

Comment:
If we set channels_min = $NUM_DMICS, we can only with channel number equal to $NUM_DMICS. In other words, we can't test with -c 2 if we set $NUM_DMICS = 4

Reply:
@bardliao what happens with -c 3 then? I dont think we want to set the min/max channels here to be 2/4. If the number of DMIC's is 4, we want to allow only 4 ch capture. @plbossart please correct me if I am wrong

Reply:
This should be defined by the min supported by the DMIC driver or based on the other settings defined as part of this config.

Reply:
that's right @ranj063, when we have 4channels we provide all 4 channels to the host. In the case of Chromebooks, the UCM file knows what channels are relevant on what platform and extract the useful channels in userspace.
Also for HDaudio platforms, we share the number of dmics with userspace, so pulseaudio/pipewire will open the PCM device with the relevant number of channels.

We currently don't have the ability (or need) to change the channel map depending on that the user asks for.

Reply:
and btw when we have 1 or 3 mics, we currently round up.

````
sof-hda-generic\;sof-hda-generic-1ch\;-DCHANNELS=2\;
sof-hda-generic\;sof-hda-generic-2ch\;-DCHANNELS=2\;
sof-hda-generic\;sof-hda-generic-3ch\;-DCHANNELS=4\;
sof-hda-generic\;sof-hda-generic-4ch\;-DCHANNELS=4\;
````